### PR TITLE
[single] Reduce wrapper memcopy for invoke

### PR DESCRIPTION
**Pull request by kparichay**
Reduced wrapper memcopy for invoke by typecasting ml_tensors_data_s->tensors to GstTensorMemory *
This reduces invoke latency

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped
    
Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Comment by myungjoo** [comment]
Please rebase and drop commits already merged and mark commits that are included in other PRs. It's hard to review when everything is mixed.

**Comment by kparichay** [comment]
Rebased.

File: api/capi/src/nnstreamer-capi-util.c [hunk 1]
```diff
@@ -600,6 +600,38 @@ ml_tensors_data_create_no_alloc (const ml_tensors_info_h info,
...
+
+  check_feature_state ();
+
+  if (data == NULL || data_src == NULL)
+    return ML_ERROR_INVALID_PARAMETER;
+
+  /* init null */
+  *data = NULL;
+
+  _data = g_new0 (ml_tensors_data_s, 1);
```
Comment (dongju-chae) [inline api/capi/src/nnstreamer-capi-util.c:621]: let's use `g_try_new0` if you wanna handle out-of-memory.

File: api/capi/src/nnstreamer-capi-single.c [hunk 2]
```diff
@@ -144,21 +149,25 @@ __setup_in_out_tensors (ml_single * single_h)
...
-  ml_tensors_data_s *in_data;
-  unsigned int i;
+  ml_tensors_data_s *in_data, *out_data;
   int status = ML_ERROR_NONE;
-  GstTensorMemory * out_tensors = single_h->out_tensors;
-  GstTensorMemory * in_tensors = single_h->in_tensors;
+  GstTensorMemory *in_tensors, *out_tensors;
+
+  status = ml_tensors_data_clone_no_alloc (&single_h->out_tensors,
+      &single_h->output);
```
Comment (dongju-chae) [inline api/capi/src/nnstreamer-capi-single.c:157]: No error handling?

Comment (kparichay) [reply api/capi/src/nnstreamer-capi-single.c:157]: Added

File: api/capi/src/nnstreamer-capi-util.c (hunk 1 above)
Comment (kparichay) [reply api/capi/src/nnstreamer-capi-util.c:621]: yes, thats a bug, updated.

**Pull request merged**